rpkg_src/crypto.cpp

```diff
@@ -1,71 +1,70 @@
 #include "crypto.h"
 
-void crypto::xor_data(char *input_data, uint32_t input_data_length)
+void crypto::xor_data(char *input_data, const uint32_t input_data_length)
 {
-    unsigned char xor_array[] = {0xDC, 0x45, 0xA6, 0x9C, 0xD3, 0x72, 0x4C, 0xAB};
-    int xor_length = sizeof(xor_array);
+    const unsigned char xor_array[] = {0xDC, 0x45, 0xA6, 0x9C, 0xD3, 0x72, 0x4C, 0xAB};
 
     for (uint64_t i = 0; i < input_data_length; i++)
     {
+        constexpr int xor_length = sizeof xor_array;
```

Comment:
Why is this in here? The size of the array will never change so it should be defined outside. I'm sure the compiler would optimise this anyway but why move it in the first place?

Reply:
Good point. Will revert later.